src/@next/Currency/Currency.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { InputProps } from '../Input/Input';
+import { StyledCurrency } from './CurrencyStyles';
+
+export type CurrencyProps = Omit<
+  InputProps,
+  'type' | 'prefix' | 'onChange' | 'value'
+> & {
+  locale?: string;
+  value?: number;
+  onChange?: (value: number) => void;
+};
+
+export const Currency = ({
+  locale = 'en',
+  value = 0,
+  onChange,
+  ...props
+}: CurrencyProps) => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const supportedLocale = Intl.ListFormat.supportedLocalesOf(locale);
+  const localeValue = supportedLocale.length > 0 ? supportedLocale[0] : 'en';
+
+  if (supportedLocale.length === 0) {
+    console.warn(`Locale value of ${locale} is unsupported, "en" will be used`);
+  }
+
+  const formatter = new Intl.NumberFormat(localeValue);
+
+  const getRawNumber = (value: string) => {
+    if (typeof value === 'number') {
+      return value;
+    }
+
+    const parts = formatter.formatToParts(1000.1);
+    const thousandSeparator = parts[1].value;
+    const decimalSeparator = parts[3].value;
+
+    const cleanedValue = parseFloat(
+      value
+        .replace(new RegExp('\\' + thousandSeparator, 'g'), '')
+        .replace(new RegExp('\\' + decimalSeparator), '.')
+    );
+
+    return Number.isNaN(cleanedValue) ? 0 : cleanedValue;
+  };
+
+  const [formattedValue, setFormattedValue] = useState(
+    formatter.format(getRawNumber(value.toString()))
+  );
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const rawValue = getRawNumber(e.currentTarget.value);
+    onChange(rawValue);
+    setFormattedValue(formatter.format(rawValue));
+  };
+
+  return (
+    <StyledCurrency
+      type="text"
+      prefix={<div>$</div>}
```

Comment:
I think we can include `currency` prop and get the correct symbol from [formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts) by providing options. based on locale

Reply:
The problem with that is currency symbol varies, and no guarantee the style will be correct. If you see the managed candidate they put the actual currency outside of the input, I think "$" is just an indicator of money

Reply:
ohh you're right, the currency is outside 👍